plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java

```diff
@@ -86,31 +85,25 @@ public DruidJdbcClient(BaseJdbcConfig config, ConnectionFactory connectionFactor
     @Override
     public Optional<JdbcTableHandle> getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        try (Connection connection = connectionFactory.openConnection(session)) {
-            String jdbcSchemaName = schemaTableName.getSchemaName();
-            String jdbcTableName = schemaTableName.getTableName();
-            try (ResultSet resultSet = getTables(connection, Optional.of(jdbcSchemaName), Optional.of(jdbcTableName))) {
-                List<JdbcTableHandle> tableHandles = new ArrayList<>();
-                while (resultSet.next()) {
+        String jdbcSchemaName = schemaTableName.getSchemaName();
+        String jdbcTableName = schemaTableName.getTableName();
+        try (
+                Connection connection = connectionFactory.openConnection(session);
+                ResultSet resultSet = getTables(connection, Optional.of(jdbcSchemaName), Optional.of(jdbcTableName))
+        ) {
+            List<JdbcTableHandle> tableHandles = new ArrayList<>();
+            while (resultSet.next()) {
+                String schemaName = resultSet.getString("TABLE_SCHEM");
+                String tableName = resultSet.getString("TABLE_NAME");
+                if (Objects.equals(schemaName, jdbcSchemaName) && Objects.equals(tableName, jdbcTableName)) {
                     tableHandles.add(new JdbcTableHandle(
                             schemaTableName,
                             DRUID_CATALOG,
-                            resultSet.getString("TABLE_SCHEM"),
-                            resultSet.getString("TABLE_NAME")));
+                            schemaName,
+                            tableName));
                 }
-                if (tableHandles.isEmpty()) {
-                    return Optional.empty();
-                }
-                return Optional.of(
-                        getOnlyElement(
-                                tableHandles
-                                        .stream()
-                                        .filter(
-                                                jdbcTableHandle ->
-                                                        Objects.equals(jdbcTableHandle.getSchemaName(), schemaTableName.getSchemaName())
-                                                                && Objects.equals(jdbcTableHandle.getTableName(), schemaTableName.getTableName()))
-                                        .collect(Collectors.toList())));
             }
+            return tableHandles.stream().findFirst();
```

Comment:
Behaviour change: if the resulting list contains more than 2 entries, the `getOnlyElement` throws `IllegalArgumentException` whereas `Stream::findFirst` just returns first entry.

Seems that there are no failing tests after change, was it intentional with `getOnlyElement `?

Reply:
Thanks for noticing this.

It's intentional. Sadly Druid connector doesn't support ANY write operations so test coverage is very limited.
This scenario is tested using BaseCaseInsensitiveMappingTests. Check the `*Clash` test methods there.

Listing and accesses are supposed to fail in case of ambiguous table or schema names.

Reply:
Thanks for confirming , @hashhar . Added explicit `verify` call.

Reply:
Can you change it to throw IllegalArgumentException instead to preserve behaviour as much as possible. I'd rather not introduce changes in untested code.

Everything else looks good.

Reply:
Putting back `getOnlyElement` to preserve behaviour.